### DISCUSSION TITLE
JTAG support with linuxgpio?

### DISCUSSION THREAD

**Discussion by fdanapfel** [avrdudes/avrdude#1494] [Ideas]
Would it be possible to use the linuxgpio driver to communicate with a target via JTAG?

All documentation I was able to find so far seems to indicate that the linuxgpio driver only supports ISP.

The use case for this would be to be able to use a "JTAG hat" on a Raspberry Pi (for example https://github.com/Blinkinlabs/JTAG_hat) to program AVR devices that only provide access via JTAG without having to buy an expensive proprietary adapter like the Atmel ICE.

openOCD apparently provides a linuxgpio driver that uses JTAG (https://openocd.org/doc-release/doxygen/linuxgpiod_8c_source.html), so maybe it is possible to reuse some of their code for avrdude for this.

**Comment by mcuee** [comment]
PR will be welcome. Thanks.
